validation/chOpenApiValidator.go

```diff
@@ -146,42 +140,116 @@ func formatError(contextId string, err error) []byte {
 	return mr
 }
 
-func extractRequestError(re *openapi3filter.RequestError, errsArray []models.CHError) []models.CHError {
+func handleMultiError(contextId string, mea *openapi3.MultiError, errsArray []models.CHError) []models.CHError {
 
-	for _, me := range re.Err.(openapi3.MultiError) {
+	// err is a *MultiError, and mea is set to the error's value, so range over it to grab each error.
+	for _, e := range *mea {
+		// Begin comparing the retrieved error (e) to possible returned error types.
 
-		// Cast to SchemaError so that we can pull out all of the necessary data to build our CH Errors response.
-		schemaError := me.(*openapi3.SchemaError)
-		reason := strings.Replace(schemaError.Reason, "\"", "'", -1)
-		jsonPath := strings.Join(schemaError.JSONPointer(), ".")
-		fieldName := schemaError.JSONPointer()[len(schemaError.JSONPointer())-1]
+		// If we have a request error, pass re into the extractRequestError func to get a formatted CHError back.
+		var re *openapi3filter.RequestError
+		if ok := errors.As(e, &re); ok {
+			errsArray = handleRequestError(contextId, re, errsArray)
+		}
 
-		// Switch over validation error for fieldValue to replace required with an empty string. Without this the
-		// error simply returns nothing when a required error is found, as it returns what the user gave (nothing).
-		fieldValue := ""
-		switch schemaError.SchemaField {
-		case "required":
-			fieldValue = ""
-		default:
-			fieldValue = fmt.Sprintf("%v", schemaError.Value)
+		// If we have a schema error, pass se into the extractSchemaError func to get a formatted CHError back.
+		var se *openapi3.SchemaError
+		if ok := errors.As(e, &se); ok {
+			errsArray = handleSchemaError(se, errsArray)
 		}
 
-		// Construct a CHError and append it to the previously created CHError slice.
-		err := models.CHError{
-			Error:        reason,
-			ErrorValues:  map[string]interface{}{fieldName: fieldValue},
-			Location:     jsonPath,
-			LocationType: "json-path",
-			Type:         "ch:validation",
+		// Between ERIC and security middleware we should never receive one of these errors
+		var sre *openapi3filter.SecurityRequirementsError
+		if ok := errors.As(e, &sre); ok {
+			log.InfoC(contextId, "Encountered unexpected security error")
 		}
+	}
+
+	// Return the populated errsArray.
+	return errsArray
+}
+
+func handleRequestError(contextId string, re *openapi3filter.RequestError, errsArray []models.CHError) []models.CHError {
+
+	// It is possible that the RequestError contains a MultiError if more than 1 error has been retuned inside of it.
+	var mea openapi3.MultiError
+	if ok := errors.As(re.Err, &mea); ok {
+		return handleMultiError(contextId, &mea, errsArray)
+	}
 
-		errsArray = append(errsArray, err)
+	// If it isn't a MultiError then we can begin the extract the error contents straight away.
+	var se *openapi3.SchemaError
+	if ok := errors.As(re.Err, &se); ok {
+		errsArray = handleSchemaError(se, errsArray)
```

Comment:
No need to pass `errsArray` into `handleSchemaError` and `handleParseError`.
These functions could just return an error which could then be appended to the array.

Reply:
Changed both functions to just return an error which is appended onto the errsArray when calling.